### PR TITLE
Adding accuracy metric classes.

### PR DESCRIPTION
**Pull request by pavithrasv**
### Summary
Adding accuracy metric classes.

### Related Issues
n/a

### PR Overview

- [y ] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/metrics.py [hunk 1]
```diff
@@ -350,7 +350,206 @@ def __init__(self, name='mean_squared_error', dtype=None):
...
+    would be .3. You can provide logits of classes as `y_pred`, since argmax of
+    logits and probabilities are same.
+
+    This metric creates two local variables, `total` and `count` that are used to
+    compute the frequency with which `y_pred` matches `y_true`. This frequency is
+    ultimately returned as `categorical accuracy`: an idempotent operation that
+    simply divides `total` by `count`.
+
+    `y_pred` and `y_true` should be passed in as vectors of probabilities, rather
+    than as labels. If necessary, use `tf.one_hot` to expand `y_true` as a vector.
```
Comment (fchollet) [inline keras/metrics.py]: `K.one_hot`

File: keras/metrics.py [hunk 2]
```diff
@@ -350,7 +350,206 @@ def __init__(self, name='mean_squared_error', dtype=None):
...
+        'sgd',
+        loss='mse',
+        metrics=[keras.metrics.CategoricalAccuracy()])
+    ```
+    """
+
+    def __init__(self, name='categorical_accuracy', dtype=None):
+        """Creates a `CategoricalAccuracy` instance.
+
+        # Arguments
```
Comment (fchollet) [inline keras/metrics.py]: Please remove the `__init__` docstrings and move the `arguments` sections to the class-level docstring (applicable everywhere)

File: keras/metrics.py [hunk 3]
```diff
@@ -350,7 +350,206 @@ def __init__(self, name='mean_squared_error', dtype=None):
...
+def accuracy(y_true, y_pred):
+    if not K.is_tensor(y_pred):
+        y_pred = K.constant(y_pred)
+    y_true = K.cast(y_true, y_pred.dtype)
+    return K.cast(K.equal(y_true, y_pred), K.floatx())
+
+
+def binary_accuracy(y_true, y_pred, threshold=0.5):
+    threshold = K.cast(threshold, y_pred.dtype)
+    y_pred = K.cast(y_pred > threshold, y_pred.dtype)
```
Comment (fchollet) [inline keras/metrics.py]: These two lines can be made conditional on `threshold != 0.5`

File: tests/keras/metrics_test.py [hunk 4]
```diff
@@ -177,6 +177,134 @@ def test_multiple_instances(self):
...
+
+        # verify that correct value is returned
+        result_t = acc_obj([[1], [0]], [[1], [0]])
+        result = K.eval(result_t)
+        assert result == 1  # 2/2
+
+        # check y_pred squeeze
+        result_t = acc_obj([[1], [1]], [[[1]], [[0]]])
+        result = K.eval(result_t)
+        assert np.isclose(result, 3 / 4, atol=1e-3)
```
Comment (fchollet) [inline tests/keras/metrics_test.py]: To avoid type issues, please use float, e.g. `3. / 4.` (test is failing with Python 2)

File: tests/keras/metrics_test.py [hunk 5]
```diff
@@ -177,6 +177,134 @@ def test_multiple_instances(self):
...
+
+        # check y_true squeeze
+        result_t = acc_obj([[[1]], [[1]]], [[1], [0]])
+        result = K.eval(result_t)
+        assert np.isclose(result, 4 / 6, atol=1e-3)
```
Comment (fchollet) [inline tests/keras/metrics_test.py]: Same here

**Comment by pavithrasv** [comment]
Thank you for the review, fixed the comments.
